src/metabase/warehouses/models/database.clj

```diff
@@ -212,6 +213,26 @@
            (log/info (u/format-color :green "Health check: success %s {:id %d}" (:name database) (:id database)))
            (analytics/inc! :metabase-database/status {:driver engine :healthy true})
 
+           ;; Detect and update provider name if not already set (only for postgres databases)
+           (when (= (name engine) "postgres")
+             (try
+               (log/info (u/format-color :blue "Provider detection: checking database %s {:id %d, :provider_name %s, :details %s}"
+                                         (:name database) (:id database) (:provider_name database)
+                                         (select-keys details [:host])))
+               (let [detected-provider (provider-detection/detect-provider-from-database
+                                        (assoc database :details details))]
+                 (log/info (u/format-color :blue "Provider detection: detected provider %s for database %s"
+                                           detected-provider (:name database)))
+                 (log/info (u/format-color :blue "Provider detection: updating %s {:id %d} from '%s' to '%s'"
+                                           (:name database) (:id database)
+                                           (:provider_name database) detected-provider))
+                 ;; Use direct query to avoid triggering search index updates
+                 (t2/query {:update :metabase_database
```

Comment:
this should just be `(t2/update! :model/Database (:id database) {:provider_name "whatever})`

Reply:
that error is unrelated